### ISSUE TITLE
"Send close" command to a remote source after 30s?

### ISSUE THREAD

**Issue by INPoppoRTUNE** [hyperion-project/hyperion.ng#1816]
- [x] I confirm that this is an issue rather than a question.

## Bug report
Hyperion return a "Send close" command to a remote source after 30s.

#### Steps to reproduce
Install [HyperTizen](https://github.com/reisxd/HyperTizen) on a Samsung TV and set it up as remote source on Hyperion.

#### What is expected?
LEDs should follow the remote input for more than 30s.

#### What is actually happening?
After 30s Hyperion return a "Send close" command to the remote source.
Please check [this thread](https://hyperion-project.org/forum/index.php?thread%2F14090-send-close-command-to-a-remote-source-after-30s%2F=#post52965) for more info.

#### System

Hyperion Server:
- Build:             (HEAD detached at 2.0.16) (Paulchen-Panther-cb85d2d/a93d79b-1705568419)
- Build time:        Jul 29 2024 19:40:01
- Git Remote:        https://github.com/hyperion-project/hyperion.ng
- Version:           2.0.16
- UI Lang:           en (BrowserLang: en-US)
- UI Access:         default
- Avail Screen Cap.: framebuffer,x11,xcb,qt
- Avail Video  Cap.: v4l2
- Avail Audio  Cap.: audio
- Avail Services:    boblight,cec,effectengine,forwarder,flatbuffer,protobuffer,mDNS,SSDP,borderdetection
- Config path:       /root/.hyperion
- Database:          read/write
- Mode:              Non-GUI

Hyperion Server OS:
- Distribution:      Ubuntu 22.04.5 LTS
- Architecture:      x86_64
- CPU Model:         Intel(R) Core(TM) i7-4790 CPU @ 3.60GHz
- CPU Type:          60
- Kernel:            linux (6.1.118-Unraid (WS: 64))
- Root/Admin:        true
- Qt Version:        6.2.4
- Python Version:    3.10.12
- Browser:           Mozilla/5.0 (X11; Linux x86_64; rv:133.0) Gecko/20100101 Firefox/133.0

**PS: this is a dockerized installation** using [foorschtbar/hyperion-docker](https://github.com/foorschtbar/hyperion-docker) image.

**Comment by Lord-Grey** [comment]
@INPoppoRTUNE I changed the WebSocket implementation.
Would you be able to test, if it fixes the issue you came across?

Via #1819 you can download a ready made Windows build, as there are no docker builds for PRs.

**Comment by ValleBL** [comment]
Can confirm this is working on the windows build
Tested for like 2-3 minutes with contant preview updates

**Comment by INPoppoRTUNE** [comment]
> [@INPoppoRTUNE](https://github.com/INPoppoRTUNE) I changed the WebSocket implementation. Would you be able to test, if it fixes the issue you came across?

Can confirm @ValleBL results, it's working.

**Comment by Lord-Grey** [comment]
Thanks all for testing it on short notice and great that it is working now!

**Comment by INPoppoRTUNE** [comment]
> Thanks all for testing it on short notice and great that it is working now!

Thank you for the quick fix!

Is there an ETA for the merge in stable?

**Comment by Lord-Grey** [comment]
There is no ETA for the next release.

**Comment by INPoppoRTUNE** [comment]
Thanks for the indicative timing, I'll try to replicate your code changes in my container in the meantime.

**Comment by ValleBL** [comment]
I created a compose.yaml using https://github.com/foorschtbar/hyperion-docker

`services:
  hyperion:
    image: foorschtbar/hyperion:nightly
    container_name: HyperionNG
    # privileged: true # for PWM
    ports:
      - 8090:8090
      - 8091:8091
      - 19444:19444
      - 19445:19445
    volumes:
      - /volume1/docker/hyperionng/config:/root/.hyperion:rw
    #devices: 
    # - /dev/video0:/dev/video0
    # - /dev/spidev0.0:/dev/spidev0.0 # for SPI
    restart: unless-stopped
    environment:
      - PUID=1026
      - PGID=100
      - TZ=Europe/Berlin
networks: {}`

The problem is that the latest (offical) nightly build does not include this PR.
Can we get at least a new nightly?

**Comment by INPoppoRTUNE** [comment]
> It includes the stuff to containerise a nightly build.

Unfortunately the Git seems to be abandoned and it's not working:
- the standalone configuration loop indefinitely, returning the following error: `/usr/bin/hyperiond: error while loading shared libraries: libpython3.9.so.1.0: cannot open shared object file: No such file or directory`
- the standard configuration return `error 8` and stops as for https://github.com/psychowood/hyperion.ng-docker/issues/6

I noticed tho that [foorschtbar/hyperion-docker](https://github.com/foorschtbar/hyperion-docker) offer a [nightly build](https://hub.docker.com/layers/foorschtbar/hyperion/nightly/images/sha256-cd09b333f8090810450f32fa59281eb2b803cd61a017d6e92c7c489dff14b12e), but I'm not sure on which Hyperion version it's based on.

Edit: nvm, @ValleBL beat me again 😅

**Comment by ValleBL** [comment]
> > It includes the stuff to containerise a nightly build.
> 
> Unfortunately the Git seems to be abandoned and it's not working:
> 
> * the standalone configuration loop indefinitely, returning the following error: `/usr/bin/hyperiond: error while loading shared libraries: libpython3.9.so.1.0: cannot open shared object file: No such file or directory`
> * the standard configuration return `error 8` and stops as for [Can’t build psychowood/hyperion.ng-docker#6](https://github.com/psychowood/hyperion.ng-docker/issues/6)
> 
> I noticed tho that [foorschtbar/hyperion-docker](https://github.com/foorschtbar/hyperion-docker) offer a [nightly build](https://hub.docker.com/layers/foorschtbar/hyperion/nightly/images/sha256-cd09b333f8090810450f32fa59281eb2b803cd61a017d6e92c7c489dff14b12e), but I'm not sure on which Hyperion version it's based on.
> 
> Edit: nvm, [@ValleBL](https://github.com/ValleBL) beat me again 😅

The repo from foorschtbar uses the latest nightly when first deployed:
https://github.com/foorschtbar/hyperion-docker/blob/f5ba08963fe48caa4aadc1944289ed11fecfcd9c/install.sh#L7

Thats why a new nightly would be great.
My stack is on version 2.0.17-beta.2+nightly202412143e7bc78449f09982ccc5af0bed211d0cda3cf68a

**Comment by Lord-Grey** [comment]
@ValleBL 

`The problem is that the latest (offical) nightly build does not include this PR.`

Are you sure? PR #1819 was merged to master 5 days ago and the master branch end in a updated nightly build afterwards.

See see the repo details

https://releases.hyperion-project.org/

**Comment by ValleBL** [comment]
I have not tested the funtionality yet, but the about page tells me build time was 14.12.24
Thats why i assumed PR #1819 is missing

```
Hyperion Server:
- Build:             (HEAD detached at 3e7bc78) (GitHub-78e1453/3e7bc78-1734126170)
- Build time:        Dec 14 2024 01:22:10
- Git Remote:        https://github.com/hyperion-project/hyperion.ng
- Version:           2.0.17-beta.2+nightly202412143e7bc78449f09982ccc5af0bed211d0cda3cf68a
- UI Lang:           de (BrowserLang: de)
- UI Access:         default
- Avail Screen Cap.: framebuffer,x11,xcb,qt
- Avail Video  Cap.: v4l2
- Avail Audio  Cap.: audio
- Avail Services:    boblight,cec,effectengine,forwarder,flatbuffer,protobuffer,mDNS,SSDP,borderdetection
- Config database:   /root/.hyperion/db/hyperion.db
- Database:          read/write
- Mode:              Non-GUI
```

**Comment by Lord-Grey** [comment]
@vallebl let me check…

**Comment by INPoppoRTUNE** [comment]
> I have not tested the funtionality yet, but the about page tells me build time was 14.12.24

I'm unable to run the nightly build.

*stable* works flawlessly:
```yaml
  hyperion:
    #image: foorschtbar/hyperion:nightly
    image: foorschtbar/hyperion:latest
    container_name: hyperion
    privileged: true # for PWM
    ports:
      - 8090:8090
      - 8091:8091
      - 19444:19444
      - 19445:19445
    volumes:
    #  - ${config}hyperion/nightly/config/:/root/.hyperion
      - ${config}hyperion/stable/config/:/root/.hyperion
    restart: unless-stopped
```

while *nightly* is unreachable at port 8090 returning `The connection to the server was reset while the page was loading.`:
```yaml
  hyperion:
    image: foorschtbar/hyperion:nightly
    #image: foorschtbar/hyperion:latest
    container_name: hyperion
    privileged: true # for PWM
    ports:
      - 8090:8090
      - 8091:8091
      - 19444:19444
      - 19445:19445
    volumes:
      - ${config}hyperion/nightly/config/:/root/.hyperion
    #  - ${config}hyperion/stable/config/:/root/.hyperion
    restart: unless-stopped
```
and the only log outputted by the container is `_lowestUpdateIntervalInSeconds:  0.005`, so I'm currently unable to verify the functionality.

**Comment by INPoppoRTUNE** [comment]
> Have you stopped the stable stack? You can use the binding only once.

Yep, I'm overwriting the container. Still no luck.

**Comment by Lord-Grey** [comment]
If you just load the ready made nightly docker image, then you will get the nighly build dated 14.12.2024 as hyperion is pulled during docker build and not when starting the docker image.

New Images build are scheduled weekly at 5:00 AM UTC every Friday (see [here](https://github.com/foorschtbar/hyperion-docker/blob/f5ba08963fe48caa4aadc1944289ed11fecfcd9c/.github/workflows/build.yml#L11))
So if you are lucky, early tomorrow morning CET you have the PR included...

**Comment by Lord-Grey** [comment]
> and the only log outputted by the container is `_lowestUpdateIntervalInSeconds: 0.005`, so I'm currently unable to verify the functionality.

If you see this message, then you have the nightly build running...

I just installed the nightly container on my Synolgy NAS and it immediately worked: 

![Image](https://github.com/user-attachments/assets/b132e512-2cdd-4f0e-a8f2-51772a77cd22)

**Comment by Lord-Grey** [comment]
There might be a problem connecting via https.
Hyperion's default configuration is to listen on port **8092** and not 8091 as configured in the docker file.

**Comment by Lord-Grey** [comment]
@INPoppoRTUNE  / @ValleBL Just updated the container image and now it includes the nightly build from today...
Which includes the WebSocket fix and other updates done since....

![Image](https://github.com/user-attachments/assets/b23ca377-5af5-49b9-882f-df7fd69e39c5)

**Comment by Lord-Grey** [comment]
@INPoppoRTUNE / @ValleBL If you do not want to wait always until Fridays for a new nightly, just log into the console of the container an do an `apt update` to get the latest nightly....

![Image](https://github.com/user-attachments/assets/392da982-9ff0-45ed-b26a-0e44694042a2)

**Comment by INPoppoRTUNE** [comment]
> [@INPoppoRTUNE](https://github.com/INPoppoRTUNE) / [@ValleBL](https://github.com/ValleBL) Just updated the container image and now it includes the nightly build from today...

Just updated the container, but it's still unreachable for me; I also tried to launch `apt update` in the container's console but it can't find any update. It's still returning `The connection to the server was reset while the page was loading.`
Can you share your whole nightly compose?

**Comment by Lord-Grey** [comment]
> Edit 2: `service` doesn't return hyperiond as running(?)

There is no service running, as Hyperion rdoes not run as a service in that container.
Hyperion is directly started via the ENTRYPOINT: /bin/sh -c "/usr/bin/hyperiond"

btw:

Are you acessing the webpage via http or https?
...and have you mapped the ports properly when running the container?

**Comment by INPoppoRTUNE** [comment]
> Then you see full debug out put in the log. That might help to see to which ports the container binds...

Here's the full logs:

```
2025-01-03T14:12:00.091 |__| EFFECTFILES       : <INFO> 41 effects loaded from directory :/effects/
2025-01-03T14:12:00.092 |__| EFFECTFILES       : <INFO> 23 effect schemas loaded from directory :/effects/schema/
2025-01-03T14:12:00.092 |__| EFFECTFILES       : <INFO> 0 effects loaded from directory /root/.hyperion/custom-effects/
2025-01-03T14:12:00.093 |__| DB                : <DEBUG> SettingsTable.cpp:278:addMissingDefaults() | Instance [0]: No missing configuration items identified
2025-01-03T14:12:00.093 |I0| BLACKBORDER       : <DEBUG> BlackBorderProcessor.cpp:67:handleSettingsUpdate() | Set mode to: default
2025-01-03T14:12:00.093 |I0| COMPONENTREG      : <DEBUG> ComponentRegister.cpp:89:setNewComponentState() | Blackborder detector: enabled
2025-01-03T14:12:00.093 |I0| IMAGETOLED        : <DEBUG> ImageProcessor.cpp:174:setReducedPixelSetFactorFactor() | Set reduced pixel set factor to 0
2025-01-03T14:12:00.093 |I0| IMAGETOLED        : <DEBUG> ImageProcessor.cpp:189:setAccuracyLevel() | Set processing accuracy level to 0
2025-01-03T14:12:00.096 |__| JSONSERVER        : <DEBUG> JsonServer.cpp:32:JsonServer() | JSON API  server created
2025-01-03T14:12:00.096 |__| SSDP              : <DEBUG> SSDPHandler.cpp:33:SSDPHandler() | SSDP info service created
2025-01-03T14:12:00.097 |I0| LEDDEVICE         : <INFO> Start LedDevice 'wled'.
2025-01-03T14:12:00.097 |I0| LEDDEVICE         : <DEBUG> LedDeviceWled.cpp:128:init() | Stream protocol   : DDP
2025-01-03T14:12:00.097 |I0| LEDDEVICE         : <DEBUG> LedDeviceWled.cpp:129:init() | Stream DDP        : 1
2025-01-03T14:12:00.097 |I0| LEDDEVICE         : <DEBUG> LedDevice.cpp:215:init() | deviceConfig: [{"autoStart":true,"brightness":255,"colorOrder":"rgb","currentLedCount":6,"enableAttempts":6,"enableAttemptsInterval":15,"hardwareLedCount":6,"host":"192.168.1.228","hostList":"NONE","latchTime":0,"overwriteBrightness":true,"overwriteSync":true,"restoreOriginalState":false,"segments":{"segmentList":"0","streamSegmentId":0,"switchOffOtherSegments":true},"stayOnAfterStreaming":false,"streamProtocol":"DDP","type":"wled"}]
2025-01-03T14:12:00.097 |I0| LEDDEVICE         : <DEBUG> LedDevice.cpp:565:setLedCount() | LedCount set to 6
2025-01-03T14:12:00.097 |I0| LEDDEVICE         : <DEBUG> LedDevice.cpp:571:setColorOrder() | ColorOrder set to RGB
2025-01-03T14:12:00.097 |I0| LEDDEVICE         : <DEBUG> LedDevice.cpp:578:setLatchTime() | LatchTime set to 0ms
2025-01-03T14:12:00.098 |I0| LEDDEVICE         : <DEBUG> LedDevice.cpp:584:setAutoStart() | AutoStart enabled
2025-01-03T14:12:00.098 |I0| LEDDEVICE         : <DEBUG> LedDevice.cpp:621:setEnableAttempts() | Max enable retries: 6, enable retry interval = 15s
2025-01-03T14:12:00.098 |I0| EFFECTENGINE      : <DEBUG> EffectEngine.cpp:128:handleUpdatedEffectList() | Effect "Candle": Add custom smoothing settings [3]. Type: Linear, Settling time: 500ms, Interval: 20Hz 
2025-01-03T14:12:00.098 |I0| LEDDEVICE         : <DEBUG> LedDeviceUdpDdp.cpp:85:init() | Hostname/IP       : 192.168.1.228
2025-01-03T14:12:00.098 |I0| LEDDEVICE         : <DEBUG> LedDeviceUdpDdp.cpp:86:init() | Port              : 4048
2025-01-03T14:12:00.098 |I0| LEDDEVICE         : <DEBUG> LedDeviceWled.cpp:150:init() | RestoreOrigState  : 0
2025-01-03T14:12:00.098 |I0| LEDDEVICE         : <DEBUG> LedDeviceWled.cpp:151:init() | Overwrite Sync.   : 1
2025-01-03T14:12:00.098 |I0| EFFECTENGINE      : <DEBUG> EffectEngine.cpp:128:handleUpdatedEffectList() | Effect "Rainbow swirl": Add custom smoothing settings [4]. Type: Linear, Settling time: 200ms, Interval: 25Hz 
2025-01-03T14:12:00.098 |I0| LEDDEVICE         : <DEBUG> LedDeviceWled.cpp:152:init() | Overwrite Brightn.: 1
2025-01-03T14:12:00.098 |I0| EFFECTENGINE      : <DEBUG> EffectEngine.cpp:128:handleUpdatedEffectList() | Effect "Random": Add custom smoothing settings [5]. Type: Linear, Settling time: 200ms, Interval: 20Hz 
2025-01-03T14:12:00.098 |I0| LEDDEVICE         : <DEBUG> LedDeviceWled.cpp:153:init() | Set Brightness to : 255
2025-01-03T14:12:00.098 |I0| EFFECTENGINE      : <DEBUG> EffectEngine.cpp:128:handleUpdatedEffectList() | Effect "Sea waves": Add custom smoothing settings [6]. Type: Linear, Settling time: 200ms, Interval: 25Hz 
2025-01-03T14:12:00.098 |I0| LEDDEVICE         : <DEBUG> LedDeviceWled.cpp:165:init() | Stream to one segment: Yes
2025-01-03T14:12:00.098 |I0| LEDDEVICE         : <DEBUG> LedDeviceWled.cpp:168:init() | Stream to segment [0]
2025-01-03T14:12:00.098 |I0| EFFECTENGINE      : <DEBUG> EffectEngine.cpp:128:handleUpdatedEffectList() | Effect "Warm mood blobs": Add custom smoothing settings [7]. Type: Linear, Settling time: 200ms, Interval: 25Hz 
2025-01-03T14:12:00.098 |I0| LEDDEVICE         : <DEBUG> LedDeviceWled.cpp:169:init() | Switch-off other segments: Yes
2025-01-03T14:12:00.098 |I0| LEDDEVICE         : <DEBUG> LedDevice.cpp:90:start() | Not enabled -> enable device
2025-01-03T14:12:00.098 |I0| LEDDEVICE         : <DEBUG> LedDevice.cpp:140:enable() | Enable device wled'
2025-01-03T14:12:00.098 |I0| LEDDEVICE         : <INFO> Stream UDP data to 192.168.1.228 port: 4048
2025-01-03T14:12:00.098 |I0| LEDDEVICE         : <INFO> Switching device wled ON
2025-01-03T14:12:00.098 |I0| BOBLIGHT          : <DEBUG> BoblightServer.cpp:31:BoblightServer() | Instance created
2025-01-03T14:12:00.098 |__| HYPERION-INSTMGR  : <INFO> Hyperion instance 'Tizen LEDS' has been started
2025-01-03T14:12:00.099 |__| JSONSERVER        : <INFO> Started on port 19444
2025-01-03T14:12:00.099 |__| WEBSERVER         : <DEBUG> WebServer.cpp:44:initServer() | Initialize http-Webserver
2025-01-03T14:12:00.099 |__| WEBSERVER         : <DEBUG> WebServer.cpp:44:initServer() | Initialize https-Webserver
2025-01-03T14:12:00.099 |__| EVENTS            : <DEBUG> EventHandler.cpp:20:EventHandler() | Hyperion event handler created
2025-01-03T14:12:00.099 |__| EVENTS-SCHED      : <DEBUG> EventScheduler.cpp:20:EventScheduler() | Hyperion event scheduler created
2025-01-03T14:12:00.099 |__| FLATBUFSERVER     : <INFO> Started on port 19400
2025-01-03T14:12:00.099 |__| PROTOSERVER       : <INFO> Started on port 19445
2025-01-03T14:12:00.099 |__| EVENTS-SCHED      : <DEBUG> EventScheduler.cpp:37:handleSettingsUpdate() | Event scheduler is enabled
2025-01-03T14:12:00.099 |__| EVENTS-SCHED      : <WARNING> No scheduled events to listen to are configured currently.
2025-01-03T14:12:00.099 |__| EVENTS-OS         : <DEBUG> OsEventHandler.cpp:44:OsEventHandlerBase() | Operating System event handler created
2025-01-03T14:12:00.100 |__| EVENTS-OS         : <INFO> The suspend/resume feature is not supported by your system configuration
2025-01-03T14:12:00.100 |__| DAEMON            : <INFO> CEC event handler created
2025-01-03T14:12:00.100 |__| DAEMON            : <INFO> FB grabber created
2025-01-03T14:12:00.100 |__| DAEMON            : <DEBUG> hyperiond.cpp:528:updateVideoGrabbers() | V4L2 grabber created
2025-01-03T14:12:00.101 |__| DAEMON            : <DEBUG> hyperiond.cpp:534:updateVideoGrabbers() | Video grabber created
2025-01-03T14:12:00.101 |__| DAEMON            : <DEBUG> hyperiond.cpp:549:updateAudioGrabbers() | Audio grabber created
2025-01-03T14:12:00.101 |__| EVENTS-CEC        : <DEBUG> CECHandler.cpp:53:handleSettingsUpdate() | CEC Event handling is enabled
2025-01-03T14:12:00.101 |__| EVENTS-CEC        : <DEBUG> CECHandler.cpp:61:handleSettingsUpdate() | Remote button press release time           : 0ms
2025-01-03T14:12:00.101 |__| EVENTS-CEC        : <DEBUG> CECHandler.cpp:62:handleSettingsUpdate() | Remote button press repeat rate            : 0ms
2025-01-03T14:12:00.101 |__| EVENTS-CEC        : <DEBUG> CECHandler.cpp:63:handleSettingsUpdate() | Remote button press delay before repeating : 200ms
2025-01-03T14:12:00.101 |__| EVENTS-CEC        : <WARNING> No CEC events to listen to are configured currently.
2025-01-03T14:12:00.108 |__| WEBSERVER         : <DEBUG> WebServer.cpp:98:handleSettingsUpdate() | Apply Webserver settings
2025-01-03T14:12:00.108 |__| WEBSERVER         : <DEBUG> WebServer.cpp:98:handleSettingsUpdate() | Apply Webserver settings
2025-01-03T14:12:00.108 |__| WEBSERVER         : <DEBUG> WebServer.cpp:118:handleSettingsUpdate() | Set document root to: :/webconfig
2025-01-03T14:12:00.108 |__| WEBSERVER         : <DEBUG> WebServer.cpp:118:handleSettingsUpdate() | Set document root to: :/webconfig
2025-01-03T14:12:00.108 |__| WEBSERVER         : <INFO> 'Hyperion http-Webserver' started on port 8090
2025-01-03T14:12:00.110 |__| WEBSERVER         : <DEBUG> WebServer.cpp:194:handleSettingsUpdate() | Setup SSL certificate
2025-01-03T14:12:00.111 |__| WEBSERVER         : <DEBUG> WebServer.cpp:212:handleSettingsUpdate() | Setup private SSL key
2025-01-03T14:12:00.111 |__| WEBSERVER         : <INFO> 'Hyperion https-Webserver' started on port 8092
2025-01-03T14:12:02.201 |I0| LEDDEVICE         : <DEBUG> ProviderRestApi.cpp:273:executeOperation() | GET took 2103ms, HTTP 0: [http://192.168.1.228/json/] []
2025-01-03T14:12:02.201 |I0| LEDDEVICE         : <ERROR> Device disabled, device 'wled' signals error: 'Retrieving device properties failed with error: 'Network request timeout error''
2025-01-03T14:12:02.201 |I0| LEDDEVICE         : <DEBUG> LedDevice.cpp:181:enable() | Device's enablement failed - Start retry timer. Retried already done [0], isEnabled: [0]
2025-01-03T14:12:02.201 |I0| LEDDEVICE         : <INFO> Start 1. attempt of 6 to enable the device in 15 seconds
2025-01-03T14:12:17.215 |I0| LEDDEVICE         : <DEBUG> LedDevice.cpp:140:enable() | Enable device wled'
2025-01-03T14:12:17.215 |I0| LEDDEVICE         : <INFO> Stream UDP data to 192.168.1.228 port: 4048
2025-01-03T14:12:17.215 |I0| LEDDEVICE         : <INFO> Switching device wled ON
2025-01-03T14:12:19.317 |I0| LEDDEVICE         : <DEBUG> ProviderRestApi.cpp:273:executeOperation() | GET took 2102ms, HTTP 0: [http://192.168.1.228/json/] []
2025-01-03T14:12:19.318 |I0| LEDDEVICE         : <ERROR> Device disabled, device 'wled' signals error: 'Retrieving device properties failed with error: 'Network request timeout error''
2025-01-03T14:12:19.318 |I0| LEDDEVICE         : <DEBUG> LedDevice.cpp:181:enable() | Device's enablement failed - Start retry timer. Retried already done [1], isEnabled: [0]
2025-01-03T14:12:19.318 |I0| LEDDEVICE         : <INFO> Start 2. attempt of 6 to enable the device in 15 seconds
2025-01-03T14:12:34.328 |I0| LEDDEVICE         : <DEBUG> LedDevice.cpp:140:enable() | Enable device wled'
2025-01-03T14:12:34.328 |I0| LEDDEVICE         : <INFO> Stream UDP data to 192.168.1.228 port: 4048
2025-01-03T14:12:34.329 |I0| LEDDEVICE         : <INFO> Switching device wled ON
2025-01-03T14:12:36.347 |I0| LEDDEVICE         : <DEBUG> ProviderRestApi.cpp:273:executeOperation() | GET took 2018ms, HTTP 0: [http://192.168.1.228/json/] []
2025-01-03T14:12:36.347 |I0| LEDDEVICE         : <ERROR> Device disabled, device 'wled' signals error: 'Retrieving device properties failed with error: 'Network request timeout error''
2025-01-03T14:12:36.347 |I0| LEDDEVICE         : <DEBUG> LedDevice.cpp:181:enable() | Device's enablement failed - Start retry timer. Retried already done [2], isEnabled: [0]
2025-01-03T14:12:36.347 |I0| LEDDEVICE         : <INFO> Start 3. attempt of 6 to enable the device in 15 seconds
2025-01-03T14:12:51.348 |I0| LEDDEVICE         : <DEBUG> LedDevice.cpp:140:enable() | Enable device wled'
2025-01-03T14:12:51.348 |I0| LEDDEVICE         : <INFO> Stream UDP data to 192.168.1.228 port: 4048
2025-01-03T14:12:51.348 |I0| LEDDEVICE         : <INFO> Switching device wled ON
2025-01-03T14:12:51.488 |I0| LEDDEVICE         : <DEBUG> ProviderRestApi.cpp:273:executeOperation() | GET took 140ms, HTTP 0: [http://192.168.1.228/json/] []
2025-01-03T14:12:51.489 |I0| LEDDEVICE         : <ERROR> Device disabled, device 'wled' signals error: 'Retrieving device properties failed with error: 'Connection refused''
2025-01-03T14:12:51.489 |I0| LEDDEVICE         : <DEBUG> LedDevice.cpp:181:enable() | Device's enablement failed - Start retry timer. Retried already done [3], isEnabled: [0]
2025-01-03T14:12:51.489 |I0| LEDDEVICE         : <INFO> Start 4. attempt of 6 to enable the device in 15 seconds
2025-01-03T14:13:06.503 |I0| LEDDEVICE         : <DEBUG> LedDevice.cpp:140:enable() | Enable device wled'
2025-01-03T14:13:06.503 |I0| LEDDEVICE         : <INFO> Stream UDP data to 192.168.1.228 port: 4048
2025-01-03T14:13:06.503 |I0| LEDDEVICE         : <INFO> Switching device wled ON
2025-01-03T14:13:06.563 |I0| LEDDEVICE         : <DEBUG> ProviderRestApi.cpp:273:executeOperation() | GET took 60ms, HTTP 200: [http://192.168.1.228/json/] []
2025-01-03T14:13:06.563 |I0| LEDDEVICE         : <DEBUG> LedDeviceWled.cpp:298:isReadyForSegmentStreaming() | Segment streaming is supported by your WLED device version [0.15.0].
2025-01-03T14:13:06.563 |I0| LEDDEVICE         : <DEBUG> LedDeviceWled.cpp:420:powerOn() | Disable synchronisation with other WLED devices
2025-01-03T14:13:06.592 |I0| LEDDEVICE         : <DEBUG> ProviderRestApi.cpp:273:executeOperation() | PUT took 29ms, HTTP 200: [http://192.168.1.228/json/state] [{"live":true,"mainseg":0,"on":true,"seg":[{"bri":255,"id":0,"on":true}],"udpn":{"recv":false,"send":false}}]
2025-01-03T14:13:06.592 |I0| LEDDEVICE         : <INFO> Device wled is ON
2025-01-03T14:13:06.592 |I0| LEDDEVICE         : <DEBUG> LedDevice.cpp:297:stopEnableAttemptsTimer() | Stopping enable retry timer
2025-01-03T14:13:06.592 |I0| LEDDEVICE         : <INFO> LedDevice 'wled' enabled
2025-01-03T14:13:06.592 |I0| COMPONENTREG      : <DEBUG> ComponentRegister.cpp:89:setNewComponentState() | LED device: enabled
```


> Plus I suggest you change
> 
> 8091:8091 to 8092:8092
> 
> to ensure https would work.

Changed, still not reachable.

**Comment by INPoppoRTUNE** [comment]
Adding `network_mode: "host"` to the compose allow the nightly container frontend to be reached.

**Comment by INPoppoRTUNE** [comment]
Not sure if strictly related, but a similar issue is back since the latest nightly update (nothing else in the config has changed):
```
2025-05-11T10:48:33.422Z [WEBSOCKET] (INFO) JSON-API service is ready to process requests
2025-05-11T10:48:33.487Z [WEBSOCKET] (DEBUG) (JsonAPI.cpp:141:handleInstanceSwitch()) Client '::ffff:192.168.1.XXX' switch to Hyperion instance 0
2025-05-11T10:49:09.279Z [JSONSERVER] (DEBUG) (JsonServer.cpp:105:newConnection()) New connection from: ::ffff:192.168.1.ZZZ
2025-05-11T10:49:09.279Z [JSONCLIENTCONNECTION] (INFO) JSON-API service is ready to process requests
2025-05-11T10:49:09.281Z [JSONSERVER] (DEBUG) (JsonServer.cpp:121:closedConnection()) Connection closed for ::ffff:192.168.1.ZZZ
2025-05-11T10:49:12.352Z [WEBSOCKET] (DEBUG) (WebSocketJsonHandler.cpp:62:onDisconnected()) WebSocket disconnected from ::ffff:192.168.1.XXX initiated via: http://192.168.1.YYY:8090
2025-05-11T10:49:12.962Z [WEBSOCKET] (DEBUG) (WebSocketJsonHandler.cpp:22:WebSocketJsonHandler()) New WebSocket connection from ::ffff:192.168.1.XXX initiated via: http://192.168.1.YYY:8090
2025-05-11T10:49:12.963Z [WEBSOCKET] (INFO) JSON-API service is ready to process requests

[...]

2025-05-11T10:53:10.399Z [JSONSERVER] (DEBUG) (JsonServer.cpp:105:newConnection()) New connection from: ::ffff:192.168.1.ZZZ
2025-05-11T10:53:10.399Z [JSONCLIENTCONNECTION] (INFO) JSON-API service is ready to process requests
2025-05-11T10:53:10.401Z [JSONSERVER] (DEBUG) (JsonServer.cpp:121:closedConnection()) Connection closed for ::ffff:192.168.1.ZZZ
```
The last three rows keep repeating indefinitely, making the JSON server unreachable.

## System Info
Hyperion Server:
- Build:             GitHub-5dba38d/bec81a8-1746455153
- Build time:        May  6 2025 01:33:47
- Build type:        Release
- Git Remote:        https://github.com/hyperion-project/hyperion.ng
- Version:           2.0.17-beta.3+nightly20250506bec81a8997efd55e1707d9d9bce1d5e3112e5bae
- UI Lang:           en (BrowserLang: en-US)
- UI Access:         default
- Avail Screen Cap.: framebuffer,x11,xcb,qt
- Avail Video  Cap.: v4l2
- Avail Audio  Cap.: audio
- Avail Services:    boblight,cec,effectengine,forwarder,flatbuffer,protobuffer,mDNS,SSDP,borderdetection
- Config database:   /root/.hyperion/db/hyperion.db
- Database:          read/write
- Mode:              Non-GUI

Hyperion Server OS:
- Distribution:      Ubuntu 22.04.5 LTS
- Architecture:      x86_64
- CPU Model:         Intel(R) Core(TM) i7-4790 CPU @ 3.60GHz
- CPU Type:          60
- Kernel:            linux (6.12.24-Unraid (WS: 64))
- Root/Admin:        true
- Qt Version:        6.2.4
- Python Version:    3.10.12
- Browser:           Mozilla/5.0 (X11; Linux x86_64; rv:138.0) Gecko/20100101 Firefox/138.0

**Comment by Lord-Grey** [comment]
The current nightly build has some issues.
You you test with PR #1860?

**Comment by Lord-Grey** [comment]
@inpopportune I merge the PR and updated the nightly build. 
Could you install the latest nightly build and check, if the issue still exists?
If yes, please open a new issue.

**Comment by INPoppoRTUNE** [comment]
> [@INPoppoRTUNE](https://github.com/INPoppoRTUNE) I merge the PR and updated the nightly build. Could you install the latest nightly build and check, if the issue still exists? If yes, please open a new issue.

I'm still in a dockerized environment, I will test it tomorrow with the fresh container:
>> New Images build are scheduled weekly at 5:00 AM UTC every Friday (see [here](https://github.com/foorschtbar/hyperion-docker/blob/f5ba08963fe48caa4aadc1944289ed11fecfcd9c/.github/workflows/build.yml#L11))

**Comment by INPoppoRTUNE** [comment]
> [@INPoppoRTUNE](https://github.com/INPoppoRTUNE) I merged the PR and updated the nightly build. Could you install the latest nightly build and check, if the issue still exists? If yes, please open a new issue.

For some reasons my server detected the update only this morning: I quickly runned a test before leaving the house and I'm now able to connect to the JSON serer and turn on and off the LEDs from external apps, so I'm guessing it's working. I'll test more in depth in the next few days, I'll report here if I find anything else.

Thanks for the quick fix!
